test/post_processors/test_collector.py

```diff
@@ -0,0 +1,282 @@
+            ),
+        ]
+
+    def coords(self):
+        return [
+            (209, 43.70009, -73.00009),
+            (210, 43.70010, -73.00010),
+            (211, 43.70011, -73.00011),
+            (212, 43.70012, -73.00012),
+            (213, 43.70013, -73.00013),
+            (214, 43.70014, -73.00014),
+            (215, 43.70015, -73.00015),
+            (216, 43.70016, -73.00016),
+            (217, 43.70017, -73.00017),
+            (218, 43.70018, -73.00018),
+            (219, 43.70019, -73.00019),
+            (220, 43.70020, -73.00020),
+            (221, 43.70021, -73.00021),
+            (222, 43.70022, -73.00022),
+            (223, 43.70023, -73.00023),
+            (224, 43.70024, -73.00024)
+        ]
+
+def test_collector_road_a():
+    collector = WayCollector(parser_class=RoadAParser)
+    collections = []
+    collector.parse('doesnt_exist.pbf', callback=lambda collection: collections.append(collection))
+    assert len(collections) == 1
+    assert len(collections[0]['ways']) == 5
+    assert collections[0]['join_type'] == 'name'
+    assert collections[0]['join_data'] == 'Road A'
+
+    expected_asc= [{'coords': [(-73.00009, 43.70009),
+                          (-73.0001, 43.7001),
+                          (-73.00011, 43.70011),
+                          (-73.00012, 43.70012)],
+               'id': 10004,
+               'refs': [9, 10, 11, 12],
+               'tags': {'highway': 'unclassified', 'name': 'Road A'}},
+              {'coords': [(-73.00012, 43.70012),
+                          (-73.00013, 43.70013),
+                          (-73.00014, 43.70014),
+                          (-73.00015, 43.70015)],
+               'id': 10003,
+               'refs': [12, 13, 14, 15],
+               'tags': {'highway': 'unclassified', 'name': 'Road A'}},
+              {'coords': [(-73.00015, 43.70015),
+                          (-73.00016, 43.70016),
+                          (-73.00017, 43.70017),
+                          (-73.00018, 43.70018)],
+               'id': 10000,
+               'refs': [15, 16, 17, 18],
+               'tags': {'highway': 'unclassified', 'name': 'Road A'}},
+              {'coords': [(-73.00018, 43.70018),
+                          (-73.00019, 43.70019),
+                          (-73.0002, 43.7002),
+                          (-73.00021, 43.70021)],
+               'id': 10001,
+               'refs': [18, 19, 20, 21],
+               'tags': {'highway': 'unclassified', 'name': 'Road A'}},
+              {'coords': [(-73.00021, 43.70021),
+                          (-73.00022, 43.70022),
+                          (-73.00023, 43.70023),
+                          (-73.00024, 43.70024)],
+                'id': 10002,
+                'refs': [21, 22, 23, 24],
+                'tags': {'highway': 'unclassified', 'name': 'Road A'}}
+        ]
+    # Joining happened in ascending order.
+    if collections[0]['ways'][0]['id'] == expected_asc[0]['id']:
```

Comment:
It's not strictly necessary, but my thought was to make the test future-proof if the algorithm for joining changes since there are two totally valid results, one in each order.